### ISSUE TITLE
Slot defined as u32 but tx TTL can be u64 on-chain, resulting in deserialisation errors

### ISSUE THREAD

**Issue by jmhrpr** [Emurgo/cardano-serialization-lib#394]
While monitoring incoming blocks I have recently encountered some transactions with TTLs larger than `u32`, which results in a deserialisation error when calling `Transaction::from_bytes(..)`. The CDDL specifies slot as `uint`. Is this a known issue?

**Comment by rooooooooob** [comment]
If you've noticed this on chain already we should probably change this. We had this appear only in pointer addresses before in #364 (fixed with #367) and I have been thinking what to do with other uses where we have things defined as `uint` in `alonzo.cddl` but in most cases they should not be that big like slot index, tx index etc but by the CBOR binary spec they're allowed to be up to 64 bits. It's a balance between easy interop with JS and not supporting deserializing these large numbers which wouldn't make sense in normal conditions vs supporting everything but going through the awkward `BigNum` interface.

**Comment by jmhrpr** [comment]
Seen minute ago in block 6968304, offending tx CBOR:

```
84A4008182582032EAAA44DC40E650F6325A01294AA93E34A3C0A87C15D2162D0A0212D9C434A600018282584C82D818584283581C8C1220953B81150FC8195FBE24E911B6140CCC9150841D67CCB4539DA101581E581CEAB1924A7C1DDF19043A6B41E0A674F1CA99AF338C91EE4B9CD2EEC1001AEFD34BEF1B00000001DCCEAEE082581D61110460F3EDD5A282531999676A137BBAD5D2F454DCBE40F7ACEF88DB1A773AA4AB021A00029075031B7FFFFFFFFFFFFFFFA10081825820C7F138CC537630C21279F3149BB51FD21797A296D3AD88EE19D076AE777D49C75840E00DF41E39235A9EB586A8939DC1B51AB6D8CB67689464CDC089F39A939C6F91E82BF93EB2CF37EE9949CCD34082A0DDEEFCDE1CC76F62CBDDF76D0531657907F5F6
```
